app/models/snp.rb

```diff
@@ -19,6 +17,34 @@ class Snp < ActiveRecord::Base
 
   after_create :default_frequencies
 
+  default_scope { select(column_names - ['genotypes']) }
+
+  def genotypes
+    genotype_ids = self.class.unscoped
+                             .select('unnest(akeys(genotypes)::int[])')
```

Comment:
Use 2 (not 25) spaces for indenting an expression in an assignment spanning multiple lines.